Fix `Input.get_joy_info()` regression after the SDL input driver PR

Nintorch opened the pull request:
While making the SDL input driver PR I didn't check what dictionary fields `Input.get_joy_info()` should return, so they went unnoticed until now.
The fields in question are `"xinput_index"`, `"raw_name"`, `"vendor_id"`, `"product_id"` and `"steam_input_index"`, and my PR also exposed an additional, essentially useless for the users `"mapping_handled"` field, that is used internally to skip Godot's gamepad mapping system (because SDL already handles that for us).
This PR should fix these issues.

*Production edit:* Fixes #110067

adamscott commented on drivers/sdl/joypad_sdl.cpp:
```diff
@@ -164,9 +165,18 @@ void JoypadSDL::process_events() {
 				Dictionary joypad_info;
-				joypad_info["mapping_handled"] = true;
+				joypad_info["mapping_handled"] = true; // Skip Godot's mapping system because SDL already handles the joypad's mapping
+				if (SDL_GetJoystickPlayerIndex(joy) >= 0) {
+					// For XInput controllers SDL_GetJoystickPlayerIndex returns the XInput user index.
+					joypad_info["xinput_index"] = itos(SDL_GetJoystickPlayerIndex(joy));
+				}
```
Should we name the key to be `player_index` instead, as it is named by SDL? I wonder if we could do like `SDL` and just return `0` on platforms that it doesn't work on.

```suggestion
				joypad_info["player_index"] = itos(SDL_GetJoystickPlayerIndex(joy));
```

adamscott replied:
Just saw that it was named like this before. Could we add both for the time, marking "xinput_index" as deprecated?

 cc. @clayjohn @mhilbrunner

clayjohn replied:
What is the benefit of renaming it? Just to match the naming in SDL?

adamscott replied:
Maybe it's not that urgent. I just felt that if SDL has a different name, then it would maybe make more sense to follow its standard.

adamscott commented on doc/classes/Input.xml:
```diff
@@ -130,16 +130,14 @@
-				[code]vendor_id[/code]: The USB vendor ID of the device.
-				[code]product_id[/code]: The USB product ID of the device.
-				On Linux:
-				[code]raw_name[/code]: The name of the controller as it came from the OS, before getting renamed by the godot controller database.
+				On Windows, Linux and macOS, the dictionary contains the following fields:
+				[code]raw_name[/code]: The name of the controller as it came from the OS, before getting renamed by the controller database.
 				[code]vendor_id[/code]: The USB vendor ID of the device.
 				[code]product_id[/code]: The USB product ID of the device.
 				[code]steam_input_index[/code]: The Steam Input gamepad index, if the device is not a Steam Input device this key won't be present.
-				[b]Note:[/b] The returned dictionary is always empty on Web, iOS, Android, and macOS.
+				On Windows, the dictionary contains an additional field:
+				[code]xinput_index[/code]: The index of the controller in the XInput system. Undefined for devices not handled by XInput.
+				[b]Note:[/b] The returned dictionary is always empty on Web, iOS, Android.
 			</description>
```
You forgot to mention about `steam_input_index`.

Nintorch replied:
Hello, I'm not sure where I forgot to mention `steam_input_index`, the top change here says it's included on Windows, Linux and macOS, and the bottom one says Windows implementation has an additional `xinput_index` field on top of the other ones 😅

adamscott reviewed (changes requested):
Finally, I found some issues.

adamscott commented:
I'm testing your PR on macOS and it seems that the dictionary returned by `Input.get_joy_info()` still empty, for some reason.

Nintorch committed:
Fix Input.get_joy_info() regression

SDL input driver did not have the "xinput_index", "raw_name", "vendor_id" and "product_id" fields for this method and exposed an additional, essentially useless for the users "mapping_handled" field. This commit fixes these issues.

Nintorch commented:
Thank you for your review, AThousandShips! I fixed the documentation according to your changes!

AThousandShips commented on doc/classes/Input.xml:
```diff
@@ -130,16 +130,14 @@
-				[b]Note:[/b] The returned dictionary is always empty on Web, iOS, Android, and macOS.
+				On Windows, the dictionary can have an additional field:
+				[code]xinput_index[/code]: The index of the controller in the XInput system. This key won't be present for devices not handled by XInput.
+				[b]Note:[/b] The returned dictionary is always empty on Android, iOS, visionOS and Web.
```
```suggestion
				[b]Note:[/b] The returned dictionary is always empty on Android, iOS, visionOS, and Web.
```

Nintorch replied:
Oops

Repiteo reviewed (approved):
Everything looks good!

Repiteo commented:
Thanks!
